consul/consul.go

```diff
@@ -1,97 +1,52 @@
 package consul
 
 import (
+	"github.com/alaa/pencil-go/registry"
 	consul "github.com/hashicorp/consul/api"
 )
 
-type ConsulClient struct {
-	client *consul.Client
+// ServiceRepository is consul-based implementation of registry.ServiceRepository
+type ServiceRepository struct {
+	consulAgent consulAgent
 }
 
-func NewConsulClient() (ConsulClient, error) {
-	client, err := consul.NewClient(consul.DefaultConfig())
-	if err != nil {
-		return ConsulClient{}, err
-	}
-	return ConsulClient{client: client}, nil
-}
-
-type ConsulAgent struct {
-	agent *consul.Agent
+type consulAgent interface {
```

Comment:
i think this interface should be upper than struct which implement this (for clear)